src/mavrosflight/mavlink_serial.cpp

```diff
@@ -135,7 +139,7 @@ void MavlinkSerial::send_message(const mavlink_message_t &msg)
   WriteBuffer *buffer = new WriteBuffer();
   buffer->len = mavlink_msg_to_send_buffer(buffer->data, &msg);
   assert(buffer->len <= MAVLINK_MAX_PACKET_LEN); //! \todo Do something less catastrophic here
-
+  // is there supposed to be a condition here?
   {
     mutex_lock lock(mutex_);
     write_queue_.push_back(buffer);
```

Comment:
Ah, it's a `scoped_mutex`.  I get why the brackets are there now.  So, the code is freezing at line 144 `mutex_lock lock(mutex_);